apps/desktop/src/test/performance.test.tsx

```
@@ -0,0 +1,524 @@
+  });
+
+  describe('Memory Usage and Cleanup', () => {
+    it('should clean up resources properly', async () => {
+      const { useMobileSessionSyncConfect } = await import('../hooks/useMobileSessionSyncConfect');
+
+      function TestCleanup() {
+        const [mounted, setMounted] = React.useState(true);
+        const [sessions, setSessions] = React.useState([]);
+        
+        if (mounted) {
+          const result = useMobileSessionSyncConfect(sessions, setSessions, true);
```

Biome (2.1.2)

[error] This hook is being called conditionally, but all hooks must be called in the exact same order in every component render.

For React to preserve state between calls, hooks needs to be called unconditionally and always in the same order.
See https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level

(lint/correctness/useHookAtTopLevel)